client-cli/src/command/transaction_command.rs

```diff
@@ -26,83 +60,219 @@ pub enum TransactionCommand {
         name: String,
+        #[structopt(name = "type", short, long, help = "Type of transaction to create")]
+        transaction_type: TransactionType,
     },
 }
 
 impl TransactionCommand {
-    pub fn execute<T: WalletClient>(&self, wallet_client: T) -> Result<()> {
+    pub fn execute<T: WalletClient, N: NetworkOpsClient>(
+        &self,
+        wallet_client: &T,
+        network_ops_client: &N,
+    ) -> Result<()> {
         match self {
-            TransactionCommand::New { chain_id, name } => {
-                Self::new_transaction(wallet_client, name, chain_id)
-            }
+            TransactionCommand::New {
+                chain_id,
+                name,
+                transaction_type,
+            } => new_transaction(
+                wallet_client,
+                network_ops_client,
+                name,
+                chain_id,
+                transaction_type,
+            ),
         }
     }
+}
 
-    fn new_transaction<T: WalletClient>(
-        wallet_client: T,
-        name: &str,
-        chain_id: &str,
-    ) -> Result<()> {
-        let passphrase = ask_passphrase()?;
-        let attributes =
-            TxAttributes::new(decode(chain_id).context(ErrorKind::DeserializationError)?[0]);
-        let outputs = Self::ask_outputs()?;
-
-        let return_address = wallet_client.new_single_transfer_address(name, &passphrase)?;
-
-        let transaction = wallet_client.create_transaction(
-            name,
-            &passphrase,
-            outputs,
-            attributes,
-            None,
-            return_address,
-        )?;
-
-        wallet_client.broadcast_transaction(&transaction)
+fn new_transaction<T: WalletClient, N: NetworkOpsClient>(
+    wallet_client: &T,
+    network_ops_client: &N,
+    name: &str,
+    chain_id: &str,
+    transaction_type: &TransactionType,
+) -> Result<()> {
+    let passphrase = ask_passphrase()?;
+
+    let transaction = match transaction_type {
+        TransactionType::Transfer => {
+            new_transfer_transaction(wallet_client, name, &passphrase, chain_id)
+        }
+        TransactionType::Deposit => {
+            new_deposit_transaction(network_ops_client, name, &passphrase, chain_id)
+        }
+        TransactionType::Unbond => {
+            new_unbond_transaction(network_ops_client, name, &passphrase, chain_id)
+        }
+        TransactionType::Withdraw => {
+            new_withdraw_transaction(network_ops_client, name, &passphrase, chain_id)
+        }
+    }?;
+
+    wallet_client.broadcast_transaction(&transaction)
+}
+
+fn new_withdraw_transaction<N: NetworkOpsClient>(
+    network_ops_client: &N,
+    name: &str,
+    passphrase: &SecUtf8,
+    chain_id: &str,
+) -> Result<TxAux> {
+    let attributes =
+        TxAttributes::new(decode(chain_id).context(ErrorKind::DeserializationError)?[0]);
+    let from_address = ask_staking_address()?;
+    let outputs = ask_outputs()?;
```

Comment:
@devashishdxt I think for the current withdraw, I guess it would be desirable:

1. to automatically put the output lock time based on `unbonded_from`
2. print some warning (or ask for confirmation) about the implicit fee (I guess the transaction type should be better called "withdraw all unbonded")

Reply:
how about entering twice for fee?

Reply:
what exactly do you have in mind for "entering twice"?
I think one solution would be a confirmation printout at the end, something like:
```
fee to be paid: <...> (minimal fee: <...>)
Do you wish to broadcast this transaction [Y/n]: 
```

Reply:
@tomtau: Is there any minimum fee for withdraw transactions? If yes, is it calculated by same `FeePolicy` as for transfer transactions?

I’m thinking of not asking for an amount from user but calculating it based on fee.